packages/piro/test/MockModelEval_A_Tpetra.cpp

```diff
@@ -349,8 +364,14 @@ void MockModelEval_A_Tpetra::evalModelImpl(
     dfdp_out->putScalar(0.0);
```

Comment:
`putScalar` runs on device, but then you write to the vector on host.  This will be quite slow.

Reply:
This is really a tiny test.

Reply:
@mperego Developers have the habit of imitating what they see -- tests get copied and pasted a lot.

Reply:
I understand. But I'm not willing to move this test on device.

Reply:
I'm not saying you need to do that.  It's just that `getDataNonConst` reserves the right to make a copy of all the Vector's device data, and copy it to host.  It doesn't do that right now because Vector uses UVM, but it could do that in the future.

Reply:
@mhoemmen  I see. I moved the getDataNonConst outside of the loops.
Can you approve if you are fine with that?